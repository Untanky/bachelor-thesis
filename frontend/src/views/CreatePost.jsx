import React, { useState } from 'react';
import { useHistory } from 'react-router-dom';
import { useSelector } from 'react-redux';
import Axios from 'axios';

import PostForm from '../components/PostForm';
import ButtonLink from '../components/ButtonLink';
import Button from '../components/Button';

import './CreatePost.scss';

const CreatePost = () => {
  const port = useSelector(state => state.languageSelector.port);
  const history = useHistory();
  const [formData, setFormData] = useState({});

  const changeForm = (key, value) => {
    setFormData({
      ...formData,
      [key]: value,
    });
  };

  const createClicked = () => {
    Axios.post(`http://localhost:${port}/api/blog/post`, formData)
      .then(() => history.push('/'))
      .catch((reason) => console.log(reason));
  }

  return (
    <div className="create-post">
      <h1 className="title">
        Create post
      </h1>
      <PostForm 
        initialState={formData}
        onChange={changeForm}
      />
      <div className="button-container">
        <ButtonLink
          type="cancel"
          to="/"
        >Cancel</ButtonLink>
        <Button 
          className="create-button"
          onClick={createClicked}
        >
          Create
        </Button>
      </div>
    </div>
  )
};

export default CreatePost;
