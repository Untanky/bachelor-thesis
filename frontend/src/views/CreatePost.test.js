import React from 'react';
import CreatePost from './CreatePost';
import renderer from 'react-test-renderer';

global.Math.random = () => 0.5;

it('renders correctly', () => {
  const tree = renderer
    .create(<CreatePost />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
