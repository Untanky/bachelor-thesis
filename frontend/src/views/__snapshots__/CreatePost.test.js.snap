// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="create-post"
>
  <h1
    className="title"
  >
    Create post
  </h1>
  <form
    className="post-form"
  >
    <div
      className="text-input"
    >
      <label
        className="label"
        htmlFor={0.5}
      >
        Title
      </label>
      <input
        className="input-field"
        defaultValue=""
        id={0.5}
        onChange={[Function]}
        placeholder="Title"
        type="text"
      />
    </div>
    <div
      className="text-input"
    >
      <label
        className="label"
        htmlFor={0.5}
      >
        Description
      </label>
      <input
        className="input-field"
        defaultValue=""
        id={0.5}
        onChange={[Function]}
        placeholder="Description"
        type="text"
      />
    </div>
  </form>
  <div
    className="button-container"
  >
    <button
      className="button cancel"
    >
      Cancel
    </button>
    <button
      className="button"
    >
      Create
    </button>
  </div>
</div>
`;
